### ISSUE TITLE
API extractor warns ES5 incompatible features in react-popover but not react-menu

### ISSUE THREAD

**Issue by ling1726** [microsoft/fluentui#18682]
#18399 Modified the tsconfig of `react-popover` to use `lib: [ES2015]` because api extract would on proxy usages in `make-styles`:

![image](https://user-images.githubusercontent.com/20744592/123130867-6f4fc080-d44d-11eb-9b98-cf9467c75498.png)

This actually does make sense becuase proxy should not be supported in ES5. The warnings don't happen from TS build, and are only triggered on api extractor.

The same issue does not happen in react-menu that was modified in the same PR with the same dependencies.

**Comment by ecraig12345** [comment]
It should be okay/desirable to change to ES2015 (or later) throughout our build once we confirm IE 11 support policy, right?

**Comment by ling1726** [comment]
Yeah, I will find some time during my shield rotation to make the IE11 support official (RFC)
